### ISSUE TITLE
Alternate icon is missing in "Included Assets" on AppStore review page

### ISSUE THREAD

**Issue by novavision** [kyubuns/AppIconChangerUnity#8]
Hi.

Most likely it's not the bug and probably I did something wrong because I never deal with alternate icons before.

So, after package installation I tried to setup scriptable object asset but get this warning by default using AutoGenerate option
![image](https://github.com/kyubuns/AppIconChangerUnity/assets/5512405/4ec3b4ad-26ab-4d2c-b9d8-1abff8f34f8f)
I am not sure what -nographics actually means and why my project has this property. Please, clarify, what's that and is there any option to force AutoGenerate option work?

Anyway, I switched to manual and referenced all required assets there. 
After build upload to the TestFlight and add that build to review, I found that alternate icon is not inclueded there
![image](https://github.com/kyubuns/AppIconChangerUnity/assets/5512405/c9a1cc34-2db3-477d-9457-1df21b8e268c)
Again, I am not 100% sure, but seems it should be there. I didn't send the build to review to ensure there is no mistake in the build.

I checked everything - alternate icon set is included to the build (checked Assets.car file), however I didn't find any entry in .plist. From what I understood it should look like this:

```<key>CFBundleIcons</key>
<dict>
    <key>CFBundlePrimaryIcon</key>
    <dict>
        <key>CFBundleIconFiles</key>
        <array>
            <string>light</string>
        </array>
        <key>UIPrerenderedIcon</key>
        <false/>
    </dict>
    <key>CFBundleAlternateIcons</key>
    <dict>
        <key>light</key>
        <dict>
            <key>CFBundleIconFiles</key>
            <array>
                <string>light</string>
            </array>
            <key>UIPrerenderedIcon</key>
            <false/>
        </dict>
        <key>dark</key>
        <dict>
            <key>CFBundleIconFiles</key>
            <array>
                <string>dark</string>
            </array>
            <key>UIPrerenderedIcon</key>
            <false/>
        </dict>
    </dict>
</dict>```

Am I wrong? Is it a bug or something else?

**Comment by novavision** [comment]
Notes for anyone who had same question:

1. new app icon shouldn't be shown in Included Assets on ap preview page
2. the .plist code I mentioned above should not exist for Unity projects. The only what should be set properly located here:
![image](https://github.com/kyubuns/AppIconChangerUnity/assets/5512405/b917a0e4-54dc-4945-a982-23931e12a03e)

But, I still have the question to @kyubuns about `Auto Generate` warning. Is there any wat to make auto generation work and what actually -nographics means?

Thanks for the great package

**Comment by kyubuns** [comment]
Thanks for the report!
I hadn't realized that I needed to edit the .plist because the dynamic changes to the icons have been working.
I will try later to see if the .plist can be updated automatically as well.

-nographics is about the parameters used to start Unity, often used in CI, etc.
https://docs.unity3d.com/2023.2/Documentation/Manual/EditorCommandLineArguments.html

**Comment by novavision** [comment]
> I hadn't realized that I needed to edit the .plist because the dynamic changes to the icons have been working.

No-no, just as I said in last message that - it not necessary should be addded to .plist file. Initially I found that .plist entry in the forum thread, but looks like it relates only on standard XCode project, not Unity exported one. So, as I know everything works fine.



> -nographics is about the parameters used to start Unity, often used in CI, etc.

Unity, as always, had the worst documentation... I didn't understand anything from the -nographics argument description. I checked command line argements in Unity Hub project settings - it's empty. Have no idea from where and why it was added automatically.

**Comment by kyubuns** [comment]
Sorry! I read it wrong!
Glad to hear the problem has been resolved.

> Unity, as always, has such a weak documentation... I didn't understand anything from the -nographics argument description. I never used batch mode. I checked command line argements in Unity Hub project settings - it's empty. Have no idea from where and why it was added automatically.

I have misled you.
This warning will always appear when you select “AutoGenerate".
The reason is that CI may build with the "-nographics" option after your work in the Editor.
If you don't use the -nographics option, don't worry about this warning.
(But I still recommend Manual Mode because it shows icons more beautifully. Auto Generate's reduction algorithm is speed-oriented and messy.)
